src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -419,6 +418,203 @@ DWORD WINAPI CProfilerManager::InstrumentationMethodThreadProc(
+    IfFailRet(pDocument->GetRootNode(&pDocumentNode));
+
+    CComPtr<CXmlNode> pCurrChildNode;
+    IfFailRet(pDocumentNode->GetChildNode(&pCurrChildNode));
+
+    CComBSTR bstrCurrChildNodeName;
+    IfFailRet(pDocumentNode->GetName(&bstrCurrChildNodeName));
+
+    if (wcscmp(bstrCurrChildNodeName, _T("InstrumentationEngineConfiguration")) != 0)
+    {
+        CLogging::LogError(_T("CConfigurationLocator::ParseAttachConfigurationThreadProc - Invalid configuration. Root element should be InstrumentationEngineConfiguration"));
+        return E_FAIL;
+    }
+
+    while (pCurrChildNode != nullptr)
+    {
+        IfFailRet(pCurrChildNode->GetName(&bstrCurrChildNodeName));
+        if (wcscmp(bstrCurrChildNodeName, _T("InstrumentationEngine")) == 0)
+        {
+            CComPtr<CXmlNode> pSettingsNode;
+            IfFailRet(pCurrChildNode->GetChildNode(&pSettingsNode));
+
+            CComBSTR bstrSettingsNodeName;
+            IfFailRet(pSettingsNode->GetName(&bstrSettingsNodeName));
+
+            if (wcscmp(bstrSettingsNodeName, _T("Settings")) == 0)
+            {
+                unordered_map<tstring, tstring> settingsMap;
+                IfFailRet(ParseSettingsConfigurationNode(pSettingsNode, settingsMap));
+
+                for (unordered_map<tstring, tstring>::iterator it = settingsMap.begin();
+                     it != settingsMap.end();
+                     it++)
+                {
+                    if (wcscmp(it->first.c_str(), _T("LogLevel")) == 0)
+                    {
+                        CLogging::SetLoggingFlags(CLoggerService::ExtractLoggingFlags(it->second.c_str()));
+                    }
+                    else if (wcscmp(it->first.c_str(), _T("LogFileLevel")) == 0)
+                    {
+                        ::SetEnvironmentVariable(_T("MicrosoftInstrumentationEngine_FileLog"), it->second.c_str());
+                    }
+                    else if (wcscmp(it->first.c_str(), _T("LogFilePath")) == 0)
+                    {
+                        ::SetEnvironmentVariable(_T("MicrosoftInstrumentationEngine_FileLogPath"), it->second.c_str());
+                    }
+                    
+                }
+            }
+        }
+        else if (wcscmp(bstrCurrChildNodeName, _T("InstrumentationMethods")) == 0)
+        {
+            CComPtr<CXmlNode> pInstrumentationMethodNode;
+            IfFailRet(pCurrChildNode->GetChildNode(&pInstrumentationMethodNode));
+
+            while (pInstrumentationMethodNode != nullptr)
+            {
+                CComBSTR bstrInstrumentationMethodNodeName;
+                IfFailRet(pInstrumentationMethodNode->GetName(&bstrInstrumentationMethodNodeName));
+
+                if (wcscmp(bstrInstrumentationMethodNodeName, _T("AddInstrumentationMethod")) == 0)
+                {
+                    CComBSTR bstrConfigPath;
+                    IfFailRet(pInstrumentationMethodNode->GetAttribute(_T("ConfigPath"), &bstrConfigPath));
```

Comment:
>GetAttribute [](start = 58, length = 12)

Can you check if GetAttribute returns an error if the attribute doesn't exist? Even if it doesn't, it's probably good to verify that bstrConfigPath is not null/empty (check it's length).